core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -333,9 +333,15 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       new SparkException("DAGScheduler cannot be initialized due to %s".format(e.getMessage))
   }
 
-  // start TaskScheduler after taskScheduler sets DAGScheduler reference in DAGScheduler's
-  // constructor
-  taskScheduler.start()
+  if (conf.getBoolean("spark.scheduler.app.slowstart", false) && master == "yarn-client") {
```

Comment:
I'm sorry that I couldn't get mesos apis to split getting applicationId and submitting resources requests to cluster master into two methods.